### ISSUE TITLE
Any plans to implement Serde?

### ISSUE THREAD

**Issue by sw-vish** [zlubsen/dis-rs#38]
Hi,
Just wanted to ask if there are any plans to implement Serde (i.e. `derive(Serialize, Deserialize)`) on the DIS PDU structs? For my particular case I'm building a web server that exposes DIS packets via an API and it'd be really helpful to be able to just serialize parsed PDUs to JSON.

**Comment by zlubsen** [comment]
Hi,
I did not give it a real thought yet, but it looks like a useful thing to do. Both for exposing via an API and pretty printing.
It can be feature-gated (eg. ```#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]```) to keep code size down when JSON is not needed.

**Comment by sw-vish** [comment]
Makes sense to me. Is that something you'd be interested in implementing yourself, or would you be open to a PR?

**Comment by zlubsen** [comment]
PRs are very welcome!

**Comment by carlocorradini** [comment]
+1

**Comment by zlubsen** [comment]
Initial work in branch 38-serde-support.

**Comment by carlocorradini** [comment]
@zlubsen Awesome 😎 

Can you rename the feature flag `serde` (easier and more standard)

**Comment by zlubsen** [comment]
1. Added derives for all PDUs and models, and a basic example.
2. Feature flag is `serde`.

Feel free to check the branch and provide comments.

**Comment by carlocorradini** [comment]
@zlubsen Super 🥳

@sw-vish Can you try it? 👍

**Comment by sw-vish** [comment]
Yep, that branch is working as expected. Thanks!

**Comment by zlubsen** [comment]
Merged to development in #58.
